pkg/operator/operator.go

```diff
@@ -770,20 +770,15 @@ func deduplicateSecrets(secretSpecs []string) []string {
 	return secretSpecs
 }
 
-func (o *Operator) ensureKotsadmApplicationMetadataConfigMap(upstreamURI string, kotsKinds *kotsutil.KotsKinds, namespace string) error {
-	kotsAppSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "Application")
-	if err != nil {
-		return errors.Wrap(err, "failed to marshal kots app spec")
-	}
-
+func (o *Operator) ensureKotsadmApplicationMetadataConfigMap(app *apptypes.App, sequence int64, namespace string, kotsKinds *kotsutil.KotsKinds, registrySettings registrytypes.RegistrySettings) error {
+	renderedKotsAppSpec, err := o.renderKotsApplicationSpec(app, sequence, namespace, kotsKinds, registrySettings, &render.Renderer{})
```

Comment:
should check the error here. my preference is to have the render logic happen inside this function since it's not really an operator related function, but i don't have a strong opinion as it is now